### PR TITLE
New input and filters

### PR DESCRIPTION
**Pull request by dquagebeur**
Hi All,
I have made a few changes/additions in my fork to meet my requirements (migration from existing logstash).
I would like to know if you would accept a pull request for all or parts ?
Features:
- add input plugin to support Azure Eventhubs
- add SCRAM support in Kafka input plugin
- improuve mutate filter (add lowercase/uppercase action and prevent rename action to create a nil value if 'from' field doesn't exist)
- add drop filter
- add 'truncate' option to stdout output plugin (diagnostic settings in azure publish an event containing an array of events, so in debug, the output is too huge)

### CODE REVIEW

File: input/azureeventhub/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+=============================
+
+## Synopsis
+
+```yaml
+input:
+  # type Must be "azureeventhub"
+  - type: azureeventhub
+
+	# connection string to azure storage account used to store consumer group offsets (required)
```
Comment (tsaikd) [inline input/azureeventhub/README.md:11]: mixing tab and space

File: input/azureeventhub/inputazureeventhub.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+		go func() {
+			for {
+				partitionClient := processor.NextPartitionClient(ctx)
+				if partitionClient == nil {
+					break
+				}
+
+				go func() {
+					wg.Add(1)
```
Comment (tsaikd) [inline input/azureeventhub/inputazureeventhub.go:110]: it's better to `defer wg.Done()` here, and some of the break in the for loop can return earlier

File: input/azureeventhub/inputazureeventhub.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+									goglog.Logger.Debugln("partition rebalanced")
+								} else {
+									goglog.Logger.Errorf("error processing partition: %v", err)
+								}
+							}
+							break
+						}
+
+						for _, event := range events {
+							_, err = t.Codec.Decode(ctx, []byte(event.Body), t.Extras, []string{}, msgChan)
```
Comment (tsaikd) [inline input/azureeventhub/inputazureeventhub.go:134]: is it reasonable to ignore all errors?

File: input/azureeventhub/inputazureeventhub.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+						}
+					}
+
+					wg.Done()
+				}()
+			}
+		}()
+
+		if err := processor.Run(ctx); err != nil {
+			panic(err)
```
Comment (tsaikd) [inline input/azureeventhub/inputazureeventhub.go:159]: it's dangerous to use `panic` here

File: input/azureeventhub/inputazureeventhub.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	return &conf, nil
+}
+
+func (t *InputConfig) Start(ctx context.Context, msgChan chan<- logevent.LogEvent) (err error) {
+
+	containerClient, err := container.NewClientFromConnectionString(t.StorageConnectionString, t.StorageContainer, nil)
+	if err != nil {
+		panic(err)
```
Comment (tsaikd) [inline input/azureeventhub/inputazureeventhub.go:68]: return err instead of panic?

Comment (tsaikd) [reply input/azureeventhub/inputazureeventhub.go]: Why not? 🤨

File: codec/azureeventhubjson/codecazureeventhubjson.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+		err = config.ErrDecodeData
+	}
+	if err != nil {
+		event.AddTag(ErrorTag)
+		goglog.Logger.Error(err)
+	}
+
+	if records, rok := event.Extra["records"]; rok && len(event.Message) == 0 {
+		for _, record := range records.([]interface{}) {
+			c.Decode(ctx, record, eventExtra, tags, msgChan)
```
Comment (tsaikd) [inline codec/azureeventhubjson/codecazureeventhubjson.go:75]: unhandled return error

File: filter/convert/filterconvert.go [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+				event.SetValue(f.To_float[0], floatVar)
+			}
+		}
+	}
+
+	// always return true here for configured filter
+	return event, true
+}
+
+func mergeField(event logevent.LogEvent, destinationName, source string) logevent.LogEvent {
```
Comment (tsaikd) [inline filter/convert/filterconvert.go:88]: unused function

**Comment by dquagebeur** [comment]
Thanks for the merge. I'll correct this last mistakes in the next PR.

**Pull request merged**